components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -452,93 +452,135 @@ private String getOIDCAuthzEndpoint(Map<String, String> authenticatorProperties)
-                String sidClaim = (String) jsonObject.get(OIDCAuthenticatorConstants.Claim.SID);
-                if (StringUtils.isNotBlank(sidClaim) && StringUtils.isNotBlank(idpName)) {
-                    // Add 'sid' claim into authentication context, to be stored in the UserSessionStore for
-                    // single logout.
-                    context.setProperty(FEDERATED_IDP_SESSION_ID + idpName, sidClaim);
-                }
+    /**
+     * Retrieves or maps the ID token according to the flow supported by the authenticator.
+     *
+     * @param context       AuthenticationContext
+     * @param request       HttpServletRequest
+     * @param oAuthResponse OAuthClientResponse
+     * @return The ID token
+     */
+    protected String mapIdToken(AuthenticationContext context, HttpServletRequest request,
+                                OAuthClientResponse oAuthResponse) {
 
-                if (log.isDebugEnabled() && IdentityUtil
-                        .isTokenLoggable(IdentityConstants.IdentityTokens.USER_ID_TOKEN)) {
-                    log.debug("Retrieved the User Information:" + jsonObject);
-                }
+        return oAuthResponse.getParam(OIDCAuthenticatorConstants.ID_TOKEN);
+    }
 
-                String authenticatedUserId = getAuthenticatedUserId(context, oAuthResponse, jsonObject);
-                String attributeSeparator = getMultiAttributeSeparator(context, authenticatedUserId);
+    /**
+     * Retrieves or maps the access token according to the flow supported by the authenticator.
+     *
+     * @param request       HttpServletRequest
+     * @param context       AuthenticationContext
+     * @param oAuthResponse OAuthClientResponse
+     * @throws AuthenticationFailedException Throws error when access token is not found
+     */
+    protected void mapAccessToken(HttpServletRequest request, AuthenticationContext context,
+                                  OAuthClientResponse oAuthResponse) throws AuthenticationFailedException {
 
-                jsonObject.entrySet().stream()
-                        .filter(entry -> !ArrayUtils.contains(NON_USER_ATTRIBUTES, entry.getKey()))
-                        .forEach(entry -> buildClaimMappings(claims, entry, attributeSeparator));
-                
-                authenticatedUser = AuthenticatedUser
-                        .createFederateAuthenticatedUserFromSubjectIdentifier(authenticatedUserId);
-            } else {
+        String accessToken = oAuthResponse.getParam(OIDCAuthenticatorConstants.ACCESS_TOKEN);
 
-                if (log.isDebugEnabled()) {
-                    log.debug("The IdToken is null");
-                }
-                authenticatedUser = AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(
-                        getAuthenticateUser(context, jsonObject, oAuthResponse));
+        if (StringUtils.isBlank(accessToken)) {
+            if (log.isDebugEnabled()) {
```

Comment:
Can we combine these two `if` blocks?

Reply:
Fixed 76bde03e8a2edc30459231f28a0cf90d9bbee240